### ISSUE TITLE
Stats Stage correction for form changes is not reflected correctly

### ISSUE THREAD

**Issue by Laioxy** [UsernameFodder/damage-eos#1]
I tested this in game and when I went to Explorer Maze and attacked Mudkip at SpD 100 with Shadow Ball on Altered Giratina at Level 100 SpA 255, the damage dealt was 131, 147, 127.
The damage was not consistent with the damage range calculated by the EoS Damage Calculator, and when I raised the defender's SpD Stage to 12, the damage was within the expected range.
I tried taking on the opponent's attack with Defence Deoxys, and it seems that my Def SpD Stage is still raised to 12 and the opponent's Atk SpA Stage is lowered to 8.

In short, I think that the Stats Stage correction due to Form is applied to both oneself and one's opponent, and the correction is reversed for the opponent.
The EoS Damage Calculator does not seem to reflect this, so I would ask you to check it once.

![POKEMONDUNGEON_SORA__706_02](https://github.com/UsernameFodder/damage-eos/assets/127853491/e76479bb-ebe1-466c-95dd-76b923018ce1)

**Comment by UsernameFodder** [comment]
I compared the code of the Japanese and North American versions, and this seems to be a bug in the Japanese version. In the Japanese version, the calculation is done like you said: for the Giratina forms, the defensive stage is modified when the _attacker_ is Giratina. In the North American version, this is fixed: the defensive stage is modified when the _defender_ is Giratina. With the Deoxys forms, the calculation is correct in both versions.

I will add a version selector to reflect the differences when I have time.

**Comment by Laioxy** [comment]
Oh, so it was a bug in the Japanese version only. Very interesting.
Let me confirm if my understanding is correct.

In the JP version, if there is a Stats Stage correction due to your form, the opponent will receive a Stats Stage correction that reverses it.
(e.g. Attacker Atk +2 -> Deffender Def -2)
In NA version, only one's own Stats Stage correction is applied.

In JP version, only Giratina's Def and SpD Stats Stage corrections are not applied due to a bug.
Strictly speaking, Def and SpD are changed only when attacking.

Is this correct?

**Comment by UsernameFodder** [comment]
> In the JP version, if there is a Stats Stage correction due to your form, the opponent will receive a Stats Stage correction that reverses it. (e.g. Attacker Atk +2 -> Deffender Def -2) In NA version, only one's own Stats Stage correction is applied.
> 
> In JP version, only Giratina's Def and SpD Stats Stage corrections are not applied due to a bug. Strictly speaking, Def and SpD are changed only when attacking.
> 
> Is this correct?

I think this is correct, but to be precise, here are the cases. I'll use the notation (attacker -> defender).

- NA
  - Giratina (atk-2) -> defender (def+0)
  - attacker (atk+0) -> Giratina (def+2)
  - Giratina-Origin (atk+2) -> defender (def+0)
  - attacker (atk+0) -> Giratina-Origin (def-2)
- JP
  - Giratina (atk-2) -> defender (def+2)
  - attacker (atk+0) -> Giratina (def+0)
  - Giratina-Origin (atk+2) -> defender (def-2)
  - attacker (atk+0) -> Giratina-Origin (def+0)

I've added a "Game version" setting to the damage calculator (a1fa84ce0f436e5a2887943e807839372f26a245), under the "Misc." section. If you set the version to "JP", it should now reflect this properly.
